jobfunnel/indeed.py

```diff
@@ -122,6 +122,123 @@ def parse_blurb(self, job, html):
+        Returns:
+            The job title scraped from soup. 
+            Note that this function may throw an AttributeError if it cannot find the title. 
+            The caller is expected to handle this exception.
+        """
+        return soup.find('a', attrs={
+            'data-tn-element': 'jobTitle'}).text.strip()
+
+    def get_company(self, soup):
+        """
+        Fetches the company from a BeautifulSoup base.
+        Args:
+			soup: BeautifulSoup base to scrape the company from.
+        Returns:
+            The company scraped from soup. 
+            Note that this function may throw an AttributeError if it cannot find the company. 
+            The caller is expected to handle this exception.
+        """
+        return soup.find('span', attrs={
+            'class': 'company'}).text.strip()
+
+    def get_location(self, soup):
+        """
+        Fetches the job location from a BeautifulSoup base.
+        Args:
+			soup: BeautifulSoup base to scrape the location from.
+        Returns:
+            The job location scraped from soup. 
+            Note that this function may throw an AttributeError if it cannot find the location. 
+            The caller is expected to handle this exception.
+        """
+        return soup.find('span', attrs={
+            'class': 'location'}).text.strip()
+
+    def get_tags(self, soup):
+        """
+        Fetches the job location from a BeautifulSoup base.
+        Args:
+			soup: BeautifulSoup base to scrape the location from.
+        Returns:
+            The job location scraped from soup. 
+            Note that this function may throw an AttributeError if it cannot find the location. 
+            The caller is expected to handle this exception.
+        """
+        table = soup.find(
+            'table', attrs={'class': 'jobCardShelfContainer'}). \
+            find_all('td', attrs={'class': 'jobCardShelfItem'})
+        return "\n".join([td.text.strip() for td in table])
+
+    def get_date(self, soup):
+        """
+        Fetches the job date from a BeautifulSoup base.
+        Args:
+			soup: BeautifulSoup base to scrape the date from.
+        Returns:
+            The job date scraped from soup. 
+            Note that this function may throw an AttributeError if it cannot find the date. 
+            The caller is expected to handle this exception.
+        """
+        return soup.find('span', attrs={
+            'class': 'date'}).text.strip()
+
+    def get_id(self, soup):
+        """
+        Fetches the job id from a BeautifulSoup base.
+        Args:
+			soup: BeautifulSoup base to scrape the id from.
+        Returns:
+            The job id scraped from soup. 
+            Note that this function may throw an AttributeError if it cannot find the id. 
+            The caller is expected to handle this exception.
+        """
+        # id regex quantifiers
+        id_regex = re.compile(r'id=\"sj_([a-zA-Z0-9]*)\"')
```

Comment:
Calling re.compile in the method ends up compiling a Regex object for every job; this object only needs to be [compiled once.](https://docs.python.org/3/library/re.html#re.compile) You should either pass the object reference as an argument or store the object in an instance attribute.